specification/logs/events/semantic_conventions/browser.md

```diff
@@ -0,0 +1,224 @@
+|---|---|---|---|---|
+| `referrer` | string | Referring Page URI (`document.referrer`) whenever available. | `https://en.wikipedia.org/wiki/Main_Page` | Recommended |
+| `type` | int | Browser page type | `0` | Required |
+| `user_consent` | boolean | Whether or not user provided consent on the page (consent for what?) |  | Recommended |
+| [`http.url`](../../../trace/semantic_conventions/http.md) | string | Full HTTP request URL in the form `scheme://host[:port]/path?query[#fragment]`. Usually the fragment is not transmitted over HTTP, but if it is known, it should be included nevertheless. [1] | `https://en.wikipedia.org/wiki/Main_Page`; `https://en.wikipedia.org/wiki/Main_Page#foo` | Required |
+
+**[1]:** The URL fragment may be included for virtual pages
+
+`type` MUST be one of the following:
+
+| Value  | Description |
+|---|---|
+| `0` | physical_page |
+| `1` | virtual_page |
+<!-- endsemconv -->
+
+## Ajax
+
+<!-- semconv browser.ajax -->
+The event name MUST be `ajax`.
+
+| Key  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `type` | int | Ajax request type | `0` | Required |
+| [`http.method`](../../../trace/semantic_conventions/http.md) | string | HTTP request method. | `GET`; `POST`; `HEAD` | Required |
+| [`http.request_content_length`](../../../trace/semantic_conventions/http.md) | int | The size of the request payload body in bytes. This is the number of bytes transferred excluding headers and is often, but not always, present as the [Content-Length](https://www.rfc-editor.org/rfc/rfc9110.html#field.content-length) header. For requests using transport encoding, this should be the compressed size. | `3495` | Recommended |
+| [`http.response_content_length`](../../../trace/semantic_conventions/http.md) | int | The size of the response payload body in bytes. This is the number of bytes transferred excluding headers and is often, but not always, present as the [Content-Length](https://www.rfc-editor.org/rfc/rfc9110.html#field.content-length) header. For requests using transport encoding, this should be the compressed size. | `3495` | Recommended |
+| [`http.status_code`](../../../trace/semantic_conventions/http.md) | int | [HTTP response status code](https://tools.ietf.org/html/rfc7231#section-6). | `200` | Conditionally Required: If and only if one was received/sent. |
+| [`http.url`](../../../trace/semantic_conventions/http.md) | string | Full HTTP request URL in the form `scheme://host[:port]/path?query[#fragment]`. Usually the fragment is not transmitted over HTTP, but if it is known, it should be included nevertheless. [1] | `https://www.foo.bar/search?q=OpenTelemetry#SemConv` | Required |
+
+**[1]:** `http.url` MUST NOT contain credentials passed via URL in form of `https://username:password@www.example.com/`. In such case the attribute's value should be `https://www.example.com/`.
+
+`type` MUST be one of the following:
+
+| Value  | Description |
+|---|---|
+| `0` | fetch |
+| `1` | xhr |
```

Comment:
We should also include (or consider)
- sendBeacon
- fetch (with keep alive) [this is an experimental and used by browsers (chromium) for seadBeacon]

Reply:
Added sendBeacon.

I read about fetch with keep-alive. It looks like keep-alive is an option on the fetch API, so it is not a different http request type but can be an additional item in event.data, say keep_alive: boolean.

Reply:
Correct, the presence of the flag does have implications during page unload as the browser will still attempt to send the request (when this flag is present) but it may be aborted when not. This is used effectively as a replacement for synchronous XHR requests (during unload) if sendBeacon() is not used. Having it as an additional property (value) would work.